blockstore/splitstore/splitstore.go

```diff
@@ -435,14 +470,18 @@ func (s *SplitStore) View(cid cid.Cid, cb func([]byte) error) error {
 	//   view can't have its data pointer deleted, which would be catastrophic.
 	//   Note that we can't just RLock for the duration of the view, as this could
 	//    lead to deadlock with recursive views.
-	wg := s.protectView(cid)
-	if wg != nil {
-		defer wg.Done()
-	}
+
+	// What happens if the transaction ends while a view is open? What happens if a transaction
+	// ends then re-opens while a view is open? Highly unlikely... but I wonder if we should
+	// just use the wait-group regardless?
+
+	// ultra-nit: you _can_ make this _slightly_ simpler by returning a possibly no-op function:
```

Comment:
a little prettier but it also allocates a closure; not sure if it is worth it.

Reply:
Well.... that can always be solved. But that just moves ugly code from one place to another, so meh.

Reply:
(really, it makes it overall worse)